django/contrib/contenttypes/views.py

```diff
@@ -18,11 +18,16 @@ def shortcut(request, content_type_id, object_id):
                 _("Content type %(ct_id)s object has no associated model")
                 % {"ct_id": content_type_id}
             )
-        obj = content_type.get_object_for_this_type(pk=object_id)
-    except (ObjectDoesNotExist, ValueError):
+        try:
+            obj = content_type.get_object_for_this_type(pk=object_id)
+        except (ObjectDoesNotExist, ValueError, ValidationError):
+            raise Http404(
+                _("Content type %(ct_id)s object %(obj_id)s doesn't exist")
+                % {"ct_id": content_type_id, "obj_id": object_id}
+            )
+    except ContentType.DoesNotExist:
```

Comment:
```suggestion
        obj = content_type.get_object_for_this_type(pk=object_id)
    except (ObjectDoesNotExist, ValueError, ValidationError):
```
the try block we already have should do 👍🏾